logicmonitor/resource_device.go

```diff
@@ -160,3 +164,50 @@ func deleteDevice(d *schema.ResourceData, m interface{}) error {
 	d.SetId("")
 	return nil
 }
+
+// function to import devices
+func resourceDeviceStateImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	client := m.(*lmv1.DefaultApi)
+
+	// if it is a deviceId, we will add the group directly
+	if checkID(d.Id()) {
+		id, err := strconv.Atoi(d.Id())
+		if err != nil {
+			return nil, err
+		}
+		restResponse, apiResponse, e := client.GetDeviceById(int32(id), "")
+		err = checkStatus(restResponse.Status, restResponse.Errmsg, apiResponse.StatusCode, apiResponse.Status, e)
+		if err != nil {
+			log.Printf("Failed to find device %q", err)
+			return nil, err
+		}
+		d.Set("ip_addr", restResponse.Data.Name)
+		d.Set("collector", restResponse.Data.PreferredCollectorId)
+		d.Set("disable_alerting", restResponse.Data.DisableAlerting)
+		d.Set("description", restResponse.Data.Description)
+		d.Set("hostgroup_id", restResponse.Data.HostGroupIds)
```

Comment:
this will probably have weird implications for dynamic groups. not sure there's really much to be done about it though.